projects/ccd-case-ui-toolkit/src/lib/shared/services/case-flag/case-flag-refdata.service.ts

```diff
@@ -24,31 +24,26 @@ export class CaseFlagRefdataService {
    */
   public getCaseFlagsRefdata(serviceId: string, flagType?: RefdataCaseFlagType, welshRequired?: boolean, externalFlagsOnly?: boolean): Observable<FlagType[]> {
     let url = this.appConfig.getCaseFlagsRefdataApiUrl();
-
     if (url) {
       url = url.replace(':sid', serviceId);
+      const requestSearchParams = new URLSearchParams();
       if (flagType) {
-        url += `?flag-type=${flagType}`;
+        requestSearchParams.set('flag-type', flagType);
       }
       if (typeof welshRequired === 'boolean') {
-        // Check if flag-type has been added to the query string; if so, append welsh-required with '&'
-        url.indexOf('?') > -1 ? url += '&' : url += '?';
-        welshRequired ? url += 'welsh-required=Y' : url += 'welsh-required=N';
-      }
-
-      if (externalFlagsOnly) {
-        // Check if anything has been added to the query string; if so, append available-external-flag with '&'
-        url.indexOf('?') > -1 ? url += '&' : url += '?';
-        url += 'available-external-flag=Y';
+        const welshFlagValue = welshRequired ? 'Y' : 'N';
+        requestSearchParams.set('welsh-required', welshFlagValue);
       }
-
+      const externalFlagsValue = externalFlagsOnly ? 'Y' : 'N';
+      requestSearchParams.set('available-external-flag', externalFlagsValue);
+      url = `${url}?${requestSearchParams.toString()}`;
```

Comment:
I guess this is OK, but if you are using URLSearchParams you should probably do:
URL foo = new URL(url)
url.searchParams.append('welsh-required', welshFlagValue)
url.searchParams.append('available-external-flag', externalFlagsValue)